openverse_catalog/dags/providers/provider_api_scripts/finnish_museums.py

```diff
@@ -1,167 +1,128 @@
-
-
-def _get_object_list(building, endpoint=ENDPOINT, retries=RETRIES):
-    page = 1
-    total_images = 0
-    condition = True
-    while condition:
-        query_params = _build_params(building=building, page=page)
-        page += 1
-        json_resp = delayed_requester.get_response_json(
-            endpoint=endpoint, retries=retries, query_params=query_params
-        )
 
-        object_list = _get_object_list_from_json(json_resp)
-        if object_list is None:
-            break
-        total_images = _process_object_list(object_list)
-
-    if total_images == 0:
-        logger.warning("No more retries ")
-        return
-
-
-def _build_params(building, default_params=None, page=1):
-    if default_params is None:
-        default_params = DEFAULT_QUERY_PARAMS
-    query_params = default_params.copy()
-    query_params.update(
-        {
-            "page": page,
-            "filter[]": [f'format:"{FORMAT_TYPE}"', f'building:"{building}"'],
-        }
-    )
-
-    return query_params
-
-
-def _get_object_list_from_json(json_resp):
-    if (
-        json_resp is None
-        or str(json_resp.get("status")).lower() != "ok"
-        or json_resp.get("records") is None
-        or len(json_resp.get("records")) == 0
-    ):
-        object_list = None
-    else:
-        object_list = json_resp.get("records")
-
-    return object_list
-
-
-def _process_object_list(object_list):
-    total_images = 0
-    if object_list is not None:
-        for obj in object_list:
-            total_images = _process_object(obj)
-
-    return total_images
-
-
-def _process_object(obj, sub_providers=SUB_PROVIDERS, provider=PROVIDER):
-    total_images = 0
-    license_url = get_license_url(obj)
-    if license_url is None:
-        return None
-    foreign_identifier = obj.get("id")
-    title = obj.get("title")
-    building = obj.get("buildings")[0].get("value")
-    source = next((s for s in sub_providers if building in sub_providers[s]), provider)
-    foreign_landing_url = _get_landing(obj)
-    raw_tags = _get_raw_tags(obj)
-    image_list = obj.get("images")
-    for img in image_list:
-        image_url = _get_image_url(img)
-        total_images = image_store.add_item(
-            license_info=get_license_info(license_url=license_url),
-            foreign_identifier=foreign_identifier,
-            foreign_landing_url=foreign_landing_url,
-            image_url=image_url,
-            title=title,
-            source=source,
-            raw_tags=raw_tags,
+class FinnishMuseumsDataIngester(ProviderDataIngester):
+    providers = {"image": PROVIDER}
+    endpoint = "https://api.finna.fi/api/v1/search"
+    batch_limit = 100
+    delay = 5
+    format_type = "0/Image/"
+
+    def ingest_records(self, **kwargs):
+        for building in BUILDINGS:
+            logger.info(f"Obtaining images of building {building}")
+            super().ingest_records(building=building)
+
+    def get_next_query_params(self, old_query_params, **kwargs):
```

Comment:
Gah, sorry I didn't recognize this in the original PR - do you think this should be `prev_query_params` instead of `old_query_params`? "Old" makes me think "legacy", which isn't quite the case here. Just a thought, not critical!

Reply:
Thank you for this -- now you mention it, I actually agree really strongly 😆 `prev` feels much better, I'll update this.